spd/run_spd.py

```diff
@@ -295,7 +294,7 @@ def optimize(
 
         # --- Evaluation --- #
         if step % config.eval_freq == 0:
-            with torch.inference_mode():
+            with torch.no_grad():
                 slow_step: bool = (
                     config.slow_eval_on_first_step
```

Comment:
Any reason for this? I don't think this case brought up by ptrblck appolies: https://discuss.pytorch.org/t/pytorch-torch-no-grad-vs-torch-inference-mode/134099?u=timgianitsos